docs/troubleshooting.md

```diff
@@ -17,6 +17,46 @@ Script analysis is provided by the [PSScriptAnalyzer] project on GitHub. If the
 message starts with `[PSScriptAnalyzer]` or if you are getting faulty script diagnostics
 (red and green squiggly lines under PowerShell in scripts) please [open an issue there].
 
+## Double-click isn't selecting the whole variable
+
+Visual Studio Code provides a default set of word separators, that is,
+characters that split words and so affect double-click selections. The editor's
+defaults include both `-` and `$`. In [v2021.5.1] we starting providing a
+default value for PowerShell files that excludes these two symbols. The
+intention of this change was to increase predictability, as double-clicking
+PowerShell symbols would now select the same portion that the extension
+highlights (with the exception of scoped variables due to the continued
+inclusion of `:` as a word separator) as well as align with collected user
+feedback.
+
+We excluded `-` because unlike programming languages that use `CamelCase` or
+`snake_case`, PowerShell uses a `Verb-Noun` style where dashes are part of many
+symbol names (like underscores in other languages). So by excluding it we teach
+Visual Studio Code to treat `Verb-Noun` as one symbol/word, which matches what
+the extension semantically highlights when the cursor is placed within it.
+
+We originally excluded `$` because PowerShell uses it as a prefix for variable
+substition, and many users were already excluding it. This change proved
+controversial and so after reviewing even more user feedback after the public
+rollout of this change, we've decided to compromise among varying user desires
+by continuing to exclude `-` but revert the exclusion of `$`.
```

Comment:
Re-reading this, I don't think we achieve much by writing this into a document &mdash; I think it comes off as defensive. Rather we should just say that there wasn't good consensus, link to the issue, and then demonstrate how to customise the setting. Anyone reading this document is trying to do the latter and our justification probably doesn't matter too much to them.

Reply:
Hm, I was trying to document reasons why many users _do_ want this configuration, and so also document a change they may wish to make. The troubleshooting guide is admittedly not the perfect place for this, but we're otherwise low on documentation resources.